core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -418,8 +418,8 @@ private[spark] object RestSubmissionClient {
   private[rest] def filterSystemEnvironment(env: Map[String, String]): Map[String, String] = {
     env.filterKeys { k =>
       // SPARK_HOME is filtered out because it is usually wrong on the remote machine (SPARK-12345)
-      (k.startsWith("SPARK_") && k != "SPARK_ENV_LOADED" && k != "SPARK_HOME") ||
-        k.startsWith("MESOS_")
+      (k.startsWith("SPARK_") && k != "SPARK_ENV_LOADED" && k != "SPARK_HOME"
+        && k != "SPARK_CONF_DIR") || k.startsWith("MESOS_")
```

Comment:
Could you add a test case in `StandaloneRestSubmitSuite.scala` in order to prevent a future regression?

Reply:
Sure, good idea. I also added a test for filtering "SPARK_HOME" which didn't get a test back when that filter was added.